packages/x-scheduler/src/joy/event-calendar/store.ts

```diff
@@ -54,21 +54,45 @@ export const selectors = {
     (state: State) => state.visibleResources,
     (events, visibleResources) => {
       const map = new Map<string, CalendarEvent[]>();
+      const allDayEventsMap = new Map<string, CalendarEvent[]>();
       for (const event of events) {
         if (event.resource && visibleResources.get(event.resource) === false) {
           continue; // Skip events for hidden resources
         }
-
         const dayKey = adapter.format(event.start, 'keyboardDate');
+        if (event.allDay) {
+          if (!allDayEventsMap.has(dayKey)) {
+            allDayEventsMap.set(dayKey, []);
+          }
+          allDayEventsMap.get(dayKey)!.push(event);
+          continue; // Only consider all-day events for this map
+        }
+
         if (!map.has(dayKey)) {
           map.set(dayKey, []);
         }
         map.get(dayKey)!.push(event);
       }
 
-      return (day: SchedulerValidDate) => {
+      return (day: SchedulerValidDate, startDateOfView?: SchedulerValidDate) => {
         const dayKey = adapter.format(day, 'keyboardDate');
-        return map.get(dayKey) || [];
+        let allDayEvents;
+        if (startDateOfView && adapter.isSameDay(day, startDateOfView)) {
+          allDayEvents = events.filter(
+            (event) =>
+              event.allDay &&
+              (!event.resource || !visibleResources.get(event.resource)) &&
+              adapter.startOfDay(event.start) <= adapter.startOfDay(day) &&
+              adapter.startOfDay(event.end) >= adapter.startOfDay(day),
+          );
+        } else {
+          allDayEvents = allDayEventsMap.get(dayKey) || [];
+        }
+
+        return {
+          regularEvents: map.get(dayKey) || [],
+          allDayEvents,
```

Comment:
Let me know what you think of this. I tried to avoid traversing all the events to create the map twice (once for regular events and once for all day)